### DISCUSSION TITLE
Is MMF supposed to work for mouse buttons 7 and 8?

### DISCUSSION THREAD

**Discussion by eternalfool** [noah-nuebling/mac-mouse-fix#130] [Questions] [answered]
I have a logitech g502 which has more buttons than the usual 5. I wanted to switch workspace using buttons 7 and 8. Just wondering if that is possible through MMF. It doesn't seem so looking at the UI which only had Middle and Side buttons.

**Comment by noah-nuebling** [comment]
Hi there, @eternalfool, thanks for reaching out!

You can download the Mac Mouse Fix 2 Beta [here](https://github.com/noah-nuebling/mac-mouse-fix/releases/download/2.0.0-Beta-11/MacMouseFixApp.zip). It works with up to 32 buttons on a mouse, so it should work with Buttons 7 and 8 on your mouse, as well.

However many Logitech mice don't send standard mouse button events on their extra buttons. If that's the case for your mouse, then the extra buttons won't work with MMF 2, either, unfortunately.

Eventually, I'd love to build a compatibility layer, so MMF can support those special buttons as well, but there's no timeframe for that, yet.

Let me know if you have any more problems or questions.

**Reply by eternalfool** [reply]
Thanks for clarifying. Looks like the events are not standard because they don't show up on MMF.
https://ibb.co/Rhb3hXj

**Reply by noah-nuebling** [reply]
Did you try the '+' button to add actions to your 7 and 8 buttons? Sorry I didn’t clarify that.

**Reply by singularity-s0** [reply]
> Did you try the '+' button to add actions to your 7 and 8 buttons? Sorry I didn’t clarify that.

Extra buttons are not recognized by MMF. I have a MX Master 2S, which has a horizontal scroll wheel and a thumb button in addition to the standard 5 buttons, but none of them is recognized.

Is it technically difficult to build such a compatibility layer? Do you need to write a driver for that or something?

**Reply by noah-nuebling** [reply]
@singularity-s0 Yeah it's somewhat difficult. As far as I can tell, you need to be able to talk to the mouse by sending raw USB reports that follow Logitechs custom protocol, and then use that to reprogram the mouse to send normal button input events. 

Let me know if you have more questions!

**Comment by noah-nuebling** [comment, marked as answer]
Hi! Marking this as answered since I'm moving the topic to Issue #356. Please continue discussion there. Thanks!
